frame/merkle-mountain-range/src/mmr/storage.rs

```diff
@@ -58,14 +58,33 @@ impl<StorageType, T, I, L> Default for Storage<StorageType, T, I, L> {
 	}
 }
 
+fn parent_hash_of_ancestor_that_added_node<T: frame_system::Config>(
+	pos: NodeIndex,
+) -> <T as frame_system::Config>::Hash {
+	// TODO: turn this in a pallet storage item recording block-num when pallet was activated.
+	let block_offset = 0;
+
+	let leaf_idx = NodesUtils::leaf_index_that_added_node(pos);
+	let block_num: <T as frame_system::Config>::BlockNumber =
+		u32::try_from(leaf_idx + block_offset)
+			.expect("leaf-idx < block-num; qed")
+			.into();
+	// TODO: I think this only holds recent history, so old block hashes might not be here.
```

Comment:
:+1:

Reply:
Have to solve this for runtime context because it's also needed on `append`ing new MMR root during block initialization (runtime context).

I will try some sort of rolling window where entries that are in this map will be saved in offchain db using `(parent_hash, leaf_index)` key and as the window progresses and they get pruned from here, mmr pallet will move them to offchain db under simple `leaf_index` key.

The key assumption here being that, in practice, forks will not be deeper than the number of blocks in this map so conflicts should never happen.